### PR TITLE
Pick kubernetes kube-gen changes

### PR DESCRIPTION
**Pull request by mfojtik**
This PR picks the refactorings that were done upstream to move the generators to kube-gen and updates the generators we use to accomodate the new import paths.

This also removes the wrappers for lister-gen and informer-gen as we can just use the upstream generator directly (I hope).

This will be a base where I need to land the kubernetes/kubernetes#49192

### CODE REVIEW

**Comment by mfojtik** [comment]
@kargakis why is openshift robot not testing this automatically?

**Comment by mfojtik** [comment]
[test]

**Comment by mfojtik** [comment]
hrm. maybe this is too much for the commit checker to swallow :-) guess I will need to do bump commit?

**Comment by kargakis** [comment]
> @kargakis why is openshift robot not testing this automatically?

Not enabled yet.

**Comment by mfojtik** [comment]
@sttts @deads2k ready for review... this should make next rebases life easier I guess and allows me to pick the https://github.com/kubernetes/kubernetes/pull/49192 and build the clients we need on top of that (finally ;-)

**Comment by deads2k** [comment]
We do some crazy-ish stuff with godep.  Take a look at godep-save.  We list out some special packages (I'm betting we'll need to do that for these eventually) and we create symlinks to staging folders from k8s.io/kubernetes.  This allows us to pick patches from the kube repo to patch our staging packages.

That's why I got openshift/kubernetes up to date for you to make it a little easier to use a hack/cherry-pick or however you wanted to do it.

**Comment by deads2k** [comment]
Ordinarily, something this size would worry me, but if we're going to pursue a separate client library, I suspect that we'll need fixes to generation that we aren't likely to backport to 1.7.

File: hack/update-generated-clientsets.sh [hunk 1]
```diff
@@ -3,7 +3,8 @@ source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
 os::build::setup_env
 
-os::util::ensure::built_binary_exists 'client-gen' 'vendor/k8s.io/kubernetes/cmd/libs/go2idl/client-gen'
+#export KUBEGEN_PKG=./vendor/k8s.io/kubernetes/vendor/k8s.io/kube-gen
```
Comment (deads2k) [inline hack/update-generated-clientsets.sh]: cruft

File: pkg/openapi/zz_generated.openapi.go [hunk 2]
```diff
@@ -12811,7 +12811,7 @@ func GetOpenAPIDefinitions(ref openapi.ReferenceCallback) map[string]openapi.Ope
 			},
 			Dependencies: []string{},
 		},
-		"k8s.io/kubernetes/cmd/libs/go2idl/client-gen/test_apis/testgroup/v1.TestType": {
+		"k8s.io/kube-gen/cmd/client-gen/test_apis/testgroup/v1.TestType": {
```
Comment (deads2k) [inline pkg/openapi/zz_generated.openapi.go]: this was generated right?

**Comment by deads2k** [comment]
squash the reactions, squash the script updates, and I would have expected generation changes.  Were there not any?

I pulled the branch locally and the issue was just github rendering weird for me

**Comment by deads2k** [comment]
nit, squash slightly smaller (categories above), make sure generation is clean, then lgtm.

File: pkg/openapi/zz_generated.openapi.go (hunk 2 above)
Comment (mfojtik) [inline pkg/openapi/zz_generated.openapi.go]: yes I dunno why :-) should I remove that test file?

**Comment by mfojtik** [comment]
@deads2k nit fixed, squashed, rebased, ready for shipping.

**Comment by deads2k** [comment]
verify failure is real.

**Comment by mfojtik** [comment]
@deads2k issue fixed, tagging till we get green tests

**Comment by deads2k** [comment]
https://github.com/openshift/origin/issues/15040

/test extended_conformance_install_update

**Comment by mfojtik** [comment]
tests are green, lets ship this

**Pull request merged**